recipes/_agent_common_service.rb

```diff
@@ -2,6 +2,15 @@
 # For windows, installing and starting service by commands, https://www.zabbix.com/documentation/3.4/manual/appendix/install/windows_agent
 # Also need to configure firewall to let Zabbix server talk to agent.
 if platform_family?('windows')
+  powershell_script 'stop_zabbix_if_exist' do
+    code <<-EOH
+    net stop 'Zabbix Agent' 2>$1 | out-null;
+    sc.exe delete "Zabbix Agent" 2>$1 | out-null;
+    Get-Process zabbix_agentd | Stop-Process -Force 2>$1 | out-null;
```

Comment:
Is the `--force` here safe? Normally on Linux you'd send a `SIGTERM` then a `SIGKILL` only if that fails, so that the process is able to clean up after itself.

Reply:
`-Force` is just for "Stops the specified processes without prompting for confirmation." 
https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/stop-process?view=powershell-6

Reply:
So this will force chef to reinstall zabbix on every chef run? Is it better to just skip the installation with a `not if`?

Reply:
@gitteri no, the `action :nothing` means this only runs if it is notified.